tests/util/localregistry/localregistry.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package localregistry
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"istio.io/istio/tests/util"
+)
+
+const (
+	// Default values for test env setup
+	LocalRegistryFile      = "tests/util/localregistry/localregistry.yaml"
+	LocalRegistryNamespace = "kube-system"
+	LocalRegistryPort      = 5000
+)
+
+// SetupLocalRegistry sets up localregistry and returns if any error was found while doing that.
+func SetupLocalRegistry(remoteRegistryPort uint16, kubeconfig string) error {
+	goPath := os.Getenv("GOPATH")
+	if len(goPath) == 0 {
+		return fmt.Errorf("GOPATH not set.")
+	}
+
+	localRegistryFilePath := path.Join(util.GetResourcePath(LocalRegistryFile))
+	if err := util.KubeApply(LocalRegistryNamespace, localRegistryFilePath, kubeconfig); err != nil {
```

Comment:
We should probably make the namespace configurable.  I suspect some of our tests will use separate namespaces for isolation.

Reply:
This namespace is defined in localregistry.yaml file itself.. do we want to change that?
Also, local registry namespace should not interfere with test namespace I think..

Reply:
Yeah, let's make that a template file so we can change the namespace.

Reply:
i'll hold off on that since, the file is used in Vagrant as well. tests/e2e/local/vagrant/Vagrantfile